src/arc_utilities/zlib_helpers.cpp

```diff
@@ -130,6 +132,30 @@ namespace ZlibHelpers
 
     std::vector<uint8_t> LoadFromFileAndDecompress(const std::string& path)
     {
+        if (!boost::filesystem::is_regular_file(path))
+        {
+            if (boost::filesystem::is_regular_file(path + "." + std::to_string(0)))
+            {
+                std::cerr << "ZLibHelpers::LoadFromFileAndDecompress(): " << path << " is not a regular file. Loading from individual slices." << std::endl;
+
+                std::vector<uint8_t> decompressed;
+                int file_idx = 0;
+                std::string slice_path = path + "." + std::to_string(file_idx);
+                do
+                {
+                    const auto slice_bytes = LoadFromFileAndDecompress(slice_path);
```

Comment:
Currently, this copies all the memory into a tmp variable, then copies it into the decompressed.
This copy could be eliminated.
(I don't think this restructuring is worth your time though, just pointing it out)